### PR TITLE
fix(cli): keep autocomplete inside short terminals

### LINKED ISSUE #2555: fix(cli): keep slash autocomplete visible in short terminals
## Problem

Slash autocomplete can render more rows than the terminal viewport, which pushes the active selection off-screen.

I reproduced this on current main (`26fb404c`) in a real 40 × 20 PTY:

```sh
stty cols 40 rows 20
XDG_CONFIG_HOME=/tmp/maka-tui-clean node packages/cli/dist/cli.js
```

Then type `/`.

The editor renders all 18 slash commands plus the editor/status chrome. The composed document is 22 rows tall, so the first two suggestion rows scroll above the 20-row viewport. The default active item is the first row (`/compact`), which means the current keyboard selection is not visible. Pressing Esc restores the screen, but while the picker is open the user cannot see what Enter will choose.

The source currently fixes `EDITOR_AUTOCOMPLETE_MAX_VISIBLE` high enough to show the whole command set. That works on normal-height terminals, but conflicts with the bottom-anchored layout on short terminals.

## Expected behavior

- Keep the active autocomplete item inside the live viewport at every supported terminal height.
- Show the full command set when space allows; otherwise use the picker's existing scrolling / more-items behavior.
- Recompute the visible-row budget when the terminal is resized while autocomplete is open.
- Keep the transcript/scrollback model unchanged.

## Suggested coverage

- Open bare `/` in a 40 × 20 fake terminal and assert that the selected row, both editor borders, and the status line are all on-screen.
- Move to the last command and assert that it becomes visible without moving the editor/status chrome.
- Resize an open autocomplete surface between short and tall terminal heights.

I found this while dogfooding the released TUI path and am happy to implement the fix.

<details>
<summary>中文说明</summary>

在 40 × 20 的真实终端中输入 `/` 时，TUI 会一次渲染全部 18 个命令，加上输入框与状态栏后共 22 行。终端只能显示最后 20 行，因此顶部两项会被推出屏幕，而默认选中的第一项 `/compact` 正好不可见；用户看不到按 Enter 将选择什么。

期望在空间足够时仍展示完整命令列表；空间不足时根据终端高度限制可见项，并使用 picker 已有的滚动能力，同时保证当前选中项、输入框和状态栏始终可见。终端在 picker 打开时发生 resize 也应重新计算高度。

我在真实 TUI 路径中复现，愿意实现该修复。

</details>

### PR DESCRIPTION
**Pull request by me2seeks**
## Summary

Slash autocomplete always rendered its full command list, so a short terminal could scroll the selected command above the live viewport while Enter still targeted that invisible item.

The TUI layout now gives autocomplete the rows left after activity, pending input, editor, and status chrome. The existing adapter crops the rendered suggestions around the current selection and preserves the upstream position counter, without rebuilding the picker or losing its selected index during resize.

Fixes #2555

<details>
<summary>中文说明</summary>

斜杠命令补全此前始终渲染完整列表。在较矮的终端中，当前选中项可能被推到可视区域上方，但按 Enter 仍会执行这个不可见的命令。

现在 TUI 会先扣除活动提示、待发送输入、编辑器和状态栏所占行数，再把剩余高度交给补全列表。现有 adapter 会围绕当前选中项裁剪渲染结果并保留位置计数；终端尺寸变化时不会重建 picker，也不会丢失选择位置。

</details>

## Verification

- `npm --workspace @maka/headless run build && npm --workspace maka-agent test` — 354/354 pass
- `npm --workspace maka-agent run typecheck`
- `npx biome check` on the five changed files
- Real 40 x 20 PTY: `/compact` and `(1/18)` remain visible on open; wrapping upward shows `/thinking` and `(18/18)`.
- Exercised live resize from 40 x 20 to 80 x 40 and back with autocomplete open; the selected item, editor borders, and status line remain visible.

## Checklist

- [x] Tests cover the change and fail without it
- [x] Lint, format, typecheck and the affected suites pass locally

Does this PR entail a change in behavior?

- [x] Yes — described under Summary above
- [ ] No

### CODE REVIEW

File: packages/cli/src/pi-tui-layout.ts [hunk 1]
```diff
@@ -95,8 +99,11 @@ export class MakaPiLayoutComponent extends Container {
     const transcriptLines = this.transcript.render(width);
     const activityLines = this.activityStrip.render(width);
     const pendingLines = this.pendingQueue.render(width);
-    const editorLines = this.editor.render(width);
     const statusLines = this.statusLine.render(width);
+    this.editor.setViewportRows(
+      this.terminal.rows - activityLines.length - pendingLines.length - statusLines.length,
```
Comment (Astro-Han) [inline packages/cli/src/pi-tui-layout.ts:124]: **[P2] Preserve one visible selection before budgeting pending chrome**

Could we make the viewport budget prioritize `selected autocomplete item + editor + status` before retaining every pending row? `pendingLines` is currently unbounded and subtracted in full, so a reachable 40×20 state with 14 queued/steering messages produces 15 pending rows; with activity, status, and the three editor-chrome rows, `viewportRows` leaves zero autocomplete rows. The picker remains active, so Enter can still execute a completely invisible selection—the original failure mode this PR is intended to remove. The clean fix seems to be clipping pending rows first and rendering an overflow summary while reserving at least one selected-item row. A focused pending-queue/short-terminal regression would capture the invariant. If an existing queue bound makes this state unreachable, please feel free to push back.

**Review by Astro-Han — APPROVED** [review]
Approved — the core approach is appropriately scoped and uses the closest available seam: it preserves the upstream picker’s selection state while fitting its rendered rows to the live terminal viewport. The focused tests cover initial clipping, wrapped selection, upstream counters, and resize behavior without duplicating authority.

I left one P2 edge case inline. It does not need to block this approval, but it can still reproduce the original invisible-selection behavior when lower-priority pending chrome consumes the entire budget. I think the clean invariant is to reserve `selected item + editor + status` first, then summarize or clip pending rows. Please feel free to push back if there is an existing queue bound or layout invariant that makes this state unreachable.

The branch was also reported as conflicting with current `main` around adjacent runner-test additions, while the production files merge cleanly. Please rebase, retain both test blocks, and let the combined CI run before merge.

<details>
<summary>中文对照</summary>

已批准。核心方案范围合理，也复用了最近的现有扩展点：保留上游 picker 的选中状态，只根据当前终端 viewport 裁剪其渲染结果。定向测试分别覆盖了首次裁剪、首尾循环选择、上游计数和 resize，没有建立重复的状态权威。

我留下了一个 P2 边界问题，不需要阻塞本次批准，但当低优先级的 pending 区域耗尽全部高度预算时，仍可能复现原 issue 中“选中项不可见但仍可执行”的行为。更干净的不变量是先为“选中项 + 编辑器 + 状态栏”保留空间，再汇总或裁剪 pending 行。如果已有队列上限或布局约束使该状态不可达，也欢迎指出。

此外，GitHub 曾报告当前分支与 `main` 在相邻新增的 runner 测试处发生冲突，生产代码本身可以自动合并。请 rebase、保留双方测试，并在合并前等待组合后的 CI 通过。

</details>

AI-assisted review disclosure: Codex assisted with diff inspection, adversarial edge-case analysis, and draft wording. Astro-Han reviewed the proposed finding, chose its severity and scope, and made the final approval decision.

File: packages/cli/src/pi-tui-layout.ts (hunk 1 above)
Comment (me2seeks) [reply packages/cli/src/pi-tui-layout.ts:124]: Fixed in 14f9c293b. Autocomplete now reserves its minimum viewport before pending rows are budgeted; excess pending entries are clipped behind an overflow summary. The focused layout test covers the 15-item queue, and the runner test confirms the selected command remains visible.

**Review by Astro-Han — APPROVED** [review]
Thanks for addressing the pending-chrome edge in `14f9c29`. The layout now reserves the autocomplete editor and one visible selected row before budgeting the pending queue, and excess pending entries collapse into a bounded summary. This closes the original invisible-selection sequence without introducing another selection authority.

The focused queue tests and runner coverage are green, as is the full CI run. At terminal heights too small to contain the fixed activity/editor/status chrome plus one suggestion, the UI can still degrade, but I consider that a separate minimum-supported-height question rather than a remaining P2 in this change.

I did not find any remaining reproducible P0–P3 issues in the supported short-terminal path. The final head is cleanly mergeable. Approved and ready to merge.

### AI-assisted review disclosure

Codex performed the incremental source, edge-case, and CI inspection. I, Astro-Han, reviewed the evidence and made the final review decision.
